pyrenew/arrayutils.py

```diff
@@ -8,6 +8,66 @@
 from jax.typing import ArrayLike
 
 
+def pad_edges_to_match(
```

Comment:
Could this be combined with `pad_to_match` to be a more flexible extension to `jnp.pad`?

Reply:
It looks like the only current use of `pad_x_to_match_y` is in `extending_pyrenew.qmd`. That use could be replaced with `pad_edges_to_match`, and `pad_x_to_match_y` could be removed.

Reply:
`pad_to_match` also appears to be unused. Let's remove `pad_to_match` and `pad_x_to_match_y` and replace with `pad_edges_to_match`.